rivalcfg/command_handlers.py

```diff
@@ -77,6 +77,58 @@ def rgbcolorshift_handler(command, colors, speed=200):
     return helpers.merge_bytes(command["command"], parsed_colors, speed)
 
 
+def rgbuniversal_handler(command, colors, positions, speed, triggers):
+    """ Returns command bytes from RGB color commands.
+    It crafts packets on a per mouse basis, by reading the format given
+    in "rgbuniversal_format" in the command dict. If a single color is given,
+    it creates a steady color command. If multiple colors are given, it creates
+    a colorshift command. If triggermask is nonzero, a reactive command is
+    created.
+
+    Arguments:
+    command -- the command description dict
+    colors -- the colors as an array of string (color name or hexadecimal RGB)
+    positions -- cumulative color positions as an array of hex strings
+    speed -- the colorshift cycle time as a string (milliseconds)
+    triggers -- trigger button mask as a hex string
+    """
+    colors = map(helpers.color_string_to_rgb, colors)
+    positions = map(lambda x: int(x, 16), positions)
+    speed = 5000 if speed == "x" else int(speed, 10)
+    triggers = 0 if triggers == "x" else int(triggers, 16)
+
+    rgb_format = command["rgbuniversal_format"]
+    header = [0] * rgb_format["header_len"]
+
+    for led_id in rgb_format["led_id"]:
+        header[led_id] = command["led_id"]
+
+    a, l = rgb_format["speed"], rgb_format["speed_len"]
+    speed = helpers.uint_to_little_endian_bytearray(speed, l)
+
+    header[a:a+l] = speed
+    header[rgb_format["repeat"]] = 1 if len(colors) == 1 or triggers > 0 else 0
+    header[rgb_format["triggers"]] = triggers
+    header[rgb_format["point_count"]] = len(colors) + 1
+
+    """ data segment format:
```

Comment:
do not add a second docstring:
* Add this to the main docstring if relevant
* else just add it as a comment :)